test/modules/TestResultCoordinator/Reports/CountingReportGenerator.cs

```diff
@@ -112,12 +117,30 @@ public async Task<ITestResultReport> CreateReportAsync()
                 hasActualResult = await this.ActualTestResults.MoveNextAsync();
             }
 
-            while (hasExpectedResult)
+            if (!this.EventHubLongHaulMode)
             {
-                totalExpectCount++;
-                TestReportUtil.EnqueueAndEnforceMaxSize(unmatchedResults, this.ExpectedTestResults.Current, this.unmatchedResultsMaxSize);
-                hasExpectedResult = await this.ExpectedTestResults.MoveNextAsync();
+                while (hasExpectedResult)
+                {
+                    totalExpectCount++;
+                    TestReportUtil.EnqueueAndEnforceMaxSize(unmatchedResults, this.ExpectedTestResults.Current, this.unmatchedResultsMaxSize);
+                    hasExpectedResult = await this.ExpectedTestResults.MoveNextAsync();
+                }
+            }
+            else
+            {
+                // If we are are using EventHub to receive messages, we see an issue where EventHub can accrue large delays after
+                // running for a while. Therefore, if we are using EventHub with this counting report, we do two things.
+                // 1. We ignore extra expected results (skipping the last hasExpectedResults loop above)
+                // 2. We make sure that the last result we got from EventHub (i.e. the lastLoadedResult) is within 1 hour.
+                //    This is an arbitrary tolerance period that we have defined, and can be tuned as needed.
+                // TODO: There is either something wrong with the EventHub service or something wrong with the way we are using it,
+                // Because we should not be accruing such large delays. If we move off EventHub, we should fix this as well.
+                if (lastLoadedResult.CreatedAt > DateTime.UtcNow - TimeSpan.FromHours(1))
```

Comment:
Is it worthwhile making this a configurable value through settings?

Reply:
You're right it could be worthwhile to make this configurable. So I changed this to be configurable through settings, but I did not actually make a change to the scripts/yamls so that we can change it from our pipeline. I did this because we are already overloaded with configurable options through the pipeline. It makes things difficult to debug. Especially for a setting like this where we will need to "tune it" in the beginning but then likely never set it again, I think it is better to have it so the user can't change it from the pipeline. If we want to change it we can easily make the change to the default value